src/common/offline_transformations/src/pruning/shrink_weights.cpp

```diff
@@ -70,6 +70,103 @@ static bool maybe_adopt_reshape_node(std::shared_ptr<ov::Node> reshape, ngraph::
     return true;
 }
 
+static bool handle_variadic_split(const std::shared_ptr<ov::Node>& split) {
+    const auto axis_node = ngraph::as_type<ngraph::opset6::Constant>(split->get_input_node_ptr(1));
+    if (!axis_node)
+        return false;
+
+    const auto& input_shape = split->get_input_partial_shape(0);
+    if (input_shape.rank().is_dynamic())
+        return false;
+
+    auto axis = axis_node->cast_vector<int64_t>()[0];
+    if (axis < 0)
+        axis += input_shape.size();
```

Comment:
If `axis` < 0 that means `axis` == -1? If so, please consider more explicit code here:
```suggestion
    if (axis == -1)
        axis = input_shape.size() - 1;
```

Reply:
axis can be also less that -1

Reply:
Could you please present an example?

Reply:
Take a look at VariadicSplit input description https://docs.openvino.ai/latest/openvino_docs_ops_movement_VariadicSplit_1.html: 
axis. Axis along data to split. A scalar or tensor with shape [1] of type T2 with value from range -rank(data) .. rank(data)-1. Negative values address dimensions from the end.